src/site/stages/build/plugins/download-drupal-assets.js

```diff
@@ -1,135 +1,50 @@
-      path.toLowerCase().endsWith(ext),
-    )
-  ) {
-    return `img/${path}`;
-  }
-
-  return `files/${path}`;
-}
-
-/*
- * Simple attribute update that replaces the url for the asset
- * with the local one
- */
-function updateAttr(attr, doc, client) {
-  const assetsToDownload = [];
-  doc(`[${attr}^="${client.getSiteUri()}/sites"]`).each((i, el) => {
-    const item = doc(el);
-    const srcAttr = item.attr(attr);
-    const newAssetPath = convertAssetPath(client.getSiteUri(), srcAttr);
-    assetsToDownload.push({
-      src: srcAttr,
-      dest: newAssetPath,
-    });
-
-    item.attr(attr, `/${newAssetPath}`);
-  });
-
-  return assetsToDownload;
-}
-
 function downloadDrupalAssets(options) {
-  const drupalClient = getDrupalClient(options);
-
   return async (files, metalsmith, done) => {
-    let assetsToDownload = [];
-
-    for (const fileName of Object.keys(files)) {
-      const file = files[fileName];
-      let fileUpdated = false;
-
-      if (file.isDrupalPage && fileName.endsWith('html')) {
-        const doc = cheerio.load(file.contents);
-
-        const srcAssets = updateAttr('src', doc, drupalClient);
-        if (srcAssets.length) fileUpdated = true;
-        assetsToDownload = assetsToDownload.concat(srcAssets);
-
-        const dataSrcAssets = updateAttr('data-src', doc, drupalClient);
-        if (dataSrcAssets.length) fileUpdated = true;
-        assetsToDownload = assetsToDownload.concat(dataSrcAssets);
-
-        const hrefAssets = updateAttr('href', doc, drupalClient);
-        if (hrefAssets.length) fileUpdated = true;
-        assetsToDownload = assetsToDownload.concat(hrefAssets);
-
-        // srset is more complicated, a requires us to parse the attribute out
-        // eslint-disable-next-line no-loop-func
-        doc(`[srcset*="${drupalClient.getSiteUri()}/sites"]`).each((i, el) => {
-          fileUpdated = true;
-          const item = doc(el);
-          const srcAttr = item.attr('srcset');
-          // srcset is a comma delimited list of srcs
-          const sources = srcAttr.split(',');
-          const newPaths = [];
-          sources.forEach(source => {
-            // Each src has a url and a size, separated by a space
-            const [url, size] = source.split(' ', 2);
-            const newAssetPath = convertAssetPath(
-              drupalClient.getSiteUri(),
-              url,
-            );
-            assetsToDownload.push({
-              src: url,
-              dest: newAssetPath,
-            });
-            newPaths.push(`/${newAssetPath} ${size}`);
-          });
-
-          item.attr('srcset', newPaths.join(','));
-        });
-
-        if (fileUpdated) {
-          file.contents = new Buffer(doc.html());
-        }
-      }
-    }
+    const assetsToDownload = Object.entries(files)
+      .filter(entry => entry[1].isDrupalAsset && !entry[1].contents)
+      .map(([key, value]) => ({
+        src: value.source,
+        dest: key,
+      }));
```

Comment:
This translation of the Drupal source to local destination took me a minute to follow, but this is really clever 🔥